File: 2021/June/Day 21/DestinationCity.py
# Source : https://leetcode.com/problems/destination-city/submissions/
class Solution(object):
    def destCity(self, paths):
        """
        :type paths: List[List[str]]
        :rtype: str
        """
        dicts ={}
        for i,j in paths:
            dicts[i]= j;

        key = paths[0][0];
        while(True):

            if(key not in dicts):
                return key;
            else:
                key = dicts[key]

File: 2023/May/Day 27/destinationCity.cpp
// Source : https://leetcode.com/problems/destination-city/submissions/
class Solution {
public:
    string destCity(vector<vector<string>>& paths) {
        map<string,string> m;
        string start="";
        int VERBOSE=0;
        for(auto&p:paths)
        {
            if(start=="")
                start=p[0];
            
            if(VERBOSE)
            cout<<p[0]<<" : "<<p[1]<<endl;
            m[p[0]]=p[1];
        }
        
        
        
        while(true)
        {
        if(size(m[start])>0)
        {
            if(VERBOSE)
            cout<<" "<<m[start];
            start=m[start];
            
        }
            else{
            break;
            }
        }
        return start;
       
    }
};